### PR TITLE
Improve IDE inspection

### PR DESCRIPTION
**Pull request by colinmollenhour**
Update is entirely vardocs. Mostly auto-generated by PHPStorm, but also added e.g. `@return Traversable|Box[]` for `BoxList#getIterator` for better IDE inspection to recognize `Box` within a foreach loop.

### CODE REVIEW

**Comment by dvdoug** [comment]
Hi @colinmollenhour 

Thanks very much for the patch!

You'll have seen the test failures from PHP CS Fixer, this is because this library is using the Symfony coding standards. These say that parameters and return types should only be included in docblocks where there is value being added - e.g. if there is also a description for the parameter. The thinking there is that since PHP7 the typing info is actually at the code level and so the docblock is therefore redundant (see https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4633).

I agree it's bit weird to see code without these sort of docblocks as for so long it's been considered a code smell not to have them!

For the case of `Traversable`, I'm happy in principle to accept a patch that gives better typing info, but I don't think `Box[]` is correct, as it's not actually an array that's returned from the function. Does something like `Traversable<Box>` work?

**Comment by colinmollenhour** [comment]
Thanks for the feedback @dvdoug!

I did see the test failures. I understand now, will just have to configure PHPStorm to ignore those inspections. :)

I tested the latest PHPStorm and unfortunately it does not seem to recognize the `Traversable<Box>` syntax. While `Box[]` may not be precisely accurate, I don't think it is entirely inaccurate either since the Iterator gives the Traversable all of the capabilities of an array. For example it is possible to write this code:

```php
$this->boxes[0]->getInnerWidth();
```

But without the `Box[]` annotation PHPStorm will flag it as an unrecognized method. IMHO adding these is a huge benefit since it fixes auto-completion and helps catch typos and other errors.

**Comment by dvdoug** [comment]
Yeah OK, the PHPDoc `Box[]` syntax is array-like, it's not promising literal `array<Box>`. So happy to accept and merge if you can remove the duplicate bits and make the build pass?

**Comment by colinmollenhour** [comment]
Fixed PHP CS Fixer pipeline and build is now passing.

File: src/PackedBox.php [hunk 1]
```diff
@@ -56,6 +56,8 @@ public function getBox(): Box
 
     /**
      * Get items packed.
+     *
+     * @return PackedItemList|PackedItem[]
```
Comment (dvdoug) [inline src/PackedBox.php:60]: This one I'm not sure about, that really is a `PackedItemList` and not a generic `Traversable`

Comment (colinmollenhour) [reply src/PackedBox.php:60]: It can probably be reverted without losing any benefit since `PackedItemList#getIterator` vardocs were also updated.

**Comment by colinmollenhour** [comment]
Fixed conflicts and removed vardoc for `getItemList`

**Comment by dvdoug** [comment]
Thanks!

**Pull request merged**